src/frontend/src/optimizer/plan_node/logical_expand.rs

```diff
@@ -263,4 +276,30 @@ mod tests {
         ];
         assert_eq!(expected_schema, values.base.schema.fields().to_owned());
     }
+
+    #[tokio::test]
+    async fn fd_derivation_expand() {
+        // input: [v1, v2, v3]
+        // FD: v1 --> { v2, v3 }
+        // output: [v1, v2, v3, flag],
+        // FD: { v1, flag } --> { v2, v3 }
```

Comment:
Different from hop & project set, I think this is correct (old FDs are lost without flag) for expand 😇